web/src/api/getWeatherData.ts

```diff
@@ -105,25 +105,28 @@ async function getWeatherData(type: WeatherType) {
   return interdata;
 }
 
-const useGetWeather = (
-  type: WeatherType,
-  options?: UseQueryOptions<Maybe<GfsForecastResponse>>
-) => {
-  return useQuery<Maybe<GfsForecastResponse>>(
-    [type],
-    async () => await getWeatherData(type),
-    {
-      staleTime: FIVE_MINUTES,
-      cacheTime: FIVE_MINUTES,
-      retry: false, // Disables retrying as getWeatherData handles retrying with new timestamps
-      ...options,
-    }
-  );
+interface UseWeatherQueryOptions
+  extends Omit<UseQueryOptions<Maybe<GfsForecastResponse>>, 'queryKey'> {
+  type: WeatherType;
+}
+
+const useGetWeather = (options: UseWeatherQueryOptions) => {
+  const { type, ...queryOptions } = options;
+  // Ensure getWeatherData returns Promise<Maybe<GfsForecastResponse>>
+  // and FIVE_MINUTES is a number representing milliseconds
```

Comment:
This comment suggest we should name the constant differently (like `FIVE_MINUTES_IN_MS`) 😅

Reply:
I mean all time in JS is in milliseconds but I guess this is what I get for asking CoPilot to document my changes...

I'll remove the comments that are not needed 😅